tools/format_python_tools.sh

```diff
@@ -15,11 +15,15 @@ echo "Running Python format check..."
 python format_python_tools.py $1
 
 echo "Running Python3 flake8 check..."
-EXCLUDE="--exclude=../benchmarks/tmp/*,*/venv/*"
+cd ..
+EXCLUDE="--exclude=benchmarks/tmp/*,*/venv/*,tools/format_python_tools.py,tools/gen_compilation_database.py,bazel-*"
 flake8 . ${EXCLUDE} --count --select=E901,E999,F821,F822,F823 --show-source --statistics
 # We raise the bar higher for benchmarks/ overall, but especially when it comes to docstrings.
 # Check everything, except indentation and line length for now.
-# Also, we ignore unused imports and redefinitions of unused, as those seems to raise false flags in test definitions.
-flake8 ../benchmarks/ ${EXCLUDE} --docstring-convention pep257 --ignore=E114,E111,E501,F401,F811 --count --show-source --statistics
+# We ignore unused imports and redefinitions of unused, as those seems to raise false flags in test definitions.
+# We ignore formatting isssues E124/E125/E126 because they conflict with the automtic fix format feature.
+flake8 . ${EXCLUDE} --ignore=E114,E111,E501,F401,F811,E124,E125,E126,D --count --show-source --statistics
```

Comment:
What is this and the following excludes excluding?

Reply:
This ignores/excludes:

- ignored, because of conflict with the automatic fix format script. This reformats according to https://github.com/envoyproxy/nighthawk/blob/master/tools/.style.yapf. We followed Envoy's lead
there, but maybe this is a good time to evaluate that:
E111 Indentation is not a multiple of four
E114 Indentation is not a multiple of four (comment) 
E501 Line too long (82 &gt; 79 characters)
E124 Closing bracket does not match visual indentation
E125 Continuation line with same indent as next logical line
E126 Continuation line over-indented for hanging indent
- False positives because of what I think are pytest peculiarities [1]. Taking a look at this again, I do think it's worth following up, because there's some actually valid complaints about unused imports too it seems. But there's also real false positives; e.g. importing pytest is something we really must do.
F401 Module imported but unused
F811 Redefinition of unused name from line n
- doc conventions
D = doc comment related checks (we do those below, checking both p257 AND google conventions as both seem to have different strengths in their checks; and so far I haven't experienced any conflicts)

[1] 
```
./test/integration/test_integration_zipkin.py:3:1: F401 'pytest' imported but unused
import pytest
^
./test/integration/test_integration_zipkin.py:7:1: F401 'integration_test_fixtures.http_test_server_fixture' imported but unused
from integration_test_fixtures import (http_test_server_fixture, server_config)
^
./test/integration/test_integration_zipkin.py:7:1: F401 'integration_test_fixtures.server_config' imported but unused
from integration_test_fixtures import (http_test_server_fixture, server_config)
^
./test/integration/test_integration_zipkin.py:11:1: F811 redefinition of unused 'http_test_server_fixture' from line 7
def test_tracing_zipkin(http_test_server_fixture):
```

Reply:
Thank you for the explanations and detailed list. I think it is generally fine for us to remain aligned with Envoy. I don't see anything that would be very concerning in the list above. This PR will significantly help on future reviews of Python code, thank you for these improvements.

Would you mind copying the explanation of the exceptions from this discussion thread into an inline comment next to the exceptions for posterity?

Reply:
Done: 1e162bc15fc8d434da3a5c859699a9de2f94bcd4